scripts/sct_process_segmentation.py

```diff
@@ -37,109 +40,138 @@ def get_parser():
+            "The eccentricity is the ratio of the focal distance (distance between focal points) over the major axis "
+            "length. The value is in the interval [0, 1). When it is 0, the ellipse becomes a circle.\n"
+            "  - orientation: angle (in degrees) between the AP axis of the spinal cord and the AP axis of the "
+            "image\n"
+            "  - solidity: CSA(spinal_cord) / CSA_convex(spinal_cord). If perfect ellipse, it should be one. This "
+            "metric is interesting for detecting non-convex shape (e.g., in case of strong compression)\n"
+            "  - length: Length of the segmentation, computed by summing the slice thickness (corrected for the "
+            "centerline angle at each slice) across the specified superior-inferior region.\n"
+        ),
+        formatter_class=SmartFormatter,
+        add_help=None,
+        prog=os.path.basename(__file__).strip(".py")
+    )
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        '-i',
+        metavar=Metavar.file,
+        required=True,
+        help="Mask to compute morphometrics from. Could be binary or weighted. E.g., spinal cord segmentation."
+             "Example: seg.nii.gz"
+    )
+
+    optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
+    optional.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        help="Show this help message and exit."
+    )
+    optional.add_argument(
+        '-o',
+        metavar=Metavar.file,
+        help="Output file name (add extension). Default: csa.csv."
+    )
+    optional.add_argument(
+        '-append',
+        choices=[0, 1],
+        default=0,
+        help="Append results as a new line in the output csv file instead of overwriting it."
+    )
+    optional.add_argument(
+        '-z',
+        metavar=Metavar.str,
+        type=str,
+        help="Slice range to compute the metrics across (requires '-p csa'). Example: 5:23"
+    )
+    optional.add_argument(
+        '-perslice',
+        choices=['0', '1'],
+        default='0',
+        help="Set to 1 to output one metric per slice instead of a single output metric. Please note that when "
+             "methods ml or map is used, outputing a single metric per slice and then averaging them all is not the "
+             "same as outputting a single metric at once across all slices."
+    )
+    optional.add_argument(
+        '-vert',
+        metavar=Metavar.str,
+        help="Vertebral levels to compute the metrics across. Example: 2:9 for C2 to T2."
+    )
+    optional.add_argument(
+        '-vertfile',
+        metavar=Metavar.str,
+        default='./label/template/PAM50_levels.nii.gz',
+        help="Vertebral labeling file. Only use with flag -vert"
+    )
+    optional.add_argument(
+        '-perlevel',
+        choices=['0', '1'],
```

Comment:
the previous implementation output the variable as an `int`:
https://github.com/neuropoly/spinalcordtoolbox/pull/2842/files#diff-8a6cdb0c20af3a3924608a180063db4eL90

Now, the variable is a str. The problem is that it is then passed in  `aggregate_per_slice_or_level()`, which expects an int (0 or 1), not a str. If str, then perlevel will be considered to be always True, causing the discrepancy observed in https://github.com/neuropoly/spinalcordtoolbox/pull/2842#issuecomment-691325349.

Conclusion:
- could you please review all the modified code that has the parameter "int" (or float) in the previous parser, to make sure that the behaviour is kept the same
- batch_processing is _very_ useful, as it helped catching that bug, otherwise difficult to catch. Tagging @Drulex here because he is currently working on adding this batch script in the CI

Reply:
Thank you for catching this. I'll start looking at this immediately.

Reply:
Excellent catch. You nailed it, @jcohenadad. Getting the expected results after fixing the two flags in that file. 

<details>
  <summary>Results</summary>

```bash
# Display results (to easily compare integrity across SCT versions)
# ===========================================================================================
set +v
~~~
Version:         git-jn/1548-msct_parser-conversion-2-398c7d0aea5dd2e1a17e0cc48a5d4bdc9fff45a9*
Ran on:          Linux pop-os 5.4.0-7642-generic
Duration:        0hrs 17min 21sec
---
t2/CSA:          73.87711295363036
mt/MTR(WM):      54.42026850022487
t2s/CSA_GM:      12.487834828856178
t2s/CSA_WM:      64.93830702088246
dmri/FA(CST_r):  0.7756557354302693
dmri/FA(CST_l):  0.7801552118509679
~~~

```
</details>

I checked and couldn't see this issue anywhere else.

Related: #2843 was created specifically to help prevent issues like this.

Reply:
Excellent catch, @jcohenadad. Fixing the two options in this file produces the expected results.

> The same issue applies to other flags (-perslice, etc.), and likely to other scripts.

This wasn't an issue in other scripts, because the original authors of these scripts were either casting the string to an int, e.g.

https://github.com/neuropoly/spinalcordtoolbox/blob/b0154f493365f893885f7b2098c21f0a097f6524/scripts/sct_warp_template.py#L154-L159

https://github.com/neuropoly/spinalcordtoolbox/blob/b0154f493365f893885f7b2098c21f0a097f6524/scripts/sct_warp_template.py#L206

Or, the authors would check for `"1"` specifically:

https://github.com/neuropoly/spinalcordtoolbox/blob/b0154f493365f893885f7b2098c21f0a097f6524/scripts/sct_propseg.py#L309-L316

https://github.com/neuropoly/spinalcordtoolbox/blob/b0154f493365f893885f7b2098c21f0a097f6524/scripts/sct_propseg.py#L590

These '0'/'1' options are handled inconsistently across all of our scripts (and not just the 10 in this PR), so I was planning on addressing that fully in #2843.

Reply:
> This is quite inconsistent, and the scope of this inconsistency goes beyond the 10 scripts in this PR, which is why I had created #2843, and was going to follow through on that after this PR was merged.

awesome! thanks a lot